core/src/main/codegen/templates/Parser.jj

```diff
@@ -7462,6 +7462,8 @@ SqlIdentifier ReservedFunctionName() :
     |   <NTILE>
     |   <NULLIF>
     |   <OCTET_LENGTH>
+    |   <PERCENTILE_CONT>
+    |   <PERCENTILE_DISC>
```

Comment:
@libenchao thanks for reviewing. I get your point. CALCITE-5564 (https://github.com/apache/calcite/pull/3210) introduced PERCENTILE_CONT & PERCENTILE_DISC as functions, it changed these 2 reserved keywords to nonReservedKeywords to let PERCENTILE_CONT & PERCENTILE_DISC can be function name without quoted backtick.

If we not put it in ReservedFunctionName(I think this is correct way, otherwise we need use backtick to quote the PERCENTILE_DISC to call the function, flink also let these 2 keywords in ReservedFunctionName), the CALCITE-5564 will be failed.

Reply:
Thanks for mentioning CALCITE-5564, it seems it extends these functions to be parsed in two ways. I'm fine with this.

Reply:
Sorry for the late comment, just to give more context: the parsing for BigQuery's PERCENTILE functions was different from that which Calcite already supported. You are correct in saying that the goal was to extend parsing to work for either form (the main difference is in operand count & presence of `WITHIN GROUP` vs `OVER` clause). It would be great if this addition could be preserved and I am happy to help review/contribute to any necessary refactoring if that's required.

Reply:
thanks tanner for sharing the background/context. If ReservedFunctionName contains this addition, PERCENTILE functions (any form) works well now.